### ISSUE TITLE
OOD

### ISSUE THREAD

**Issue by liuxuexun** [ByteDance-Seed/Bagel#172]
感谢开源您的工作，我在运行sft脚本的时候，在[`deepcopy`](https://github.com/ByteDance-Seed/Bagel/blob/c8d5e35f7d87d043b748caa87c45a946caf3265a/train/pretrain_unified_navit.py#L482C7-L482C32) 这行直接爆内存了，我有800G的内存，请问是什么原因造成的呢？

以下是我的运行脚本：
```
  torchrun \
  --nnodes=1 \
  --node_rank=0 \
  --nproc_per_node=8 \
  train/pretrain_unified_navit.py \
  --dataset_config_file ./data/configs/example.yaml \
  --model_path /root/model/bagel \
  --layer_module Qwen2MoTDecoderLayer \
  --max_latent_size 64 \
  --finetune_from_hf True \
  --auto_resume False \
  --resume-model-only False \
  --finetune-from-ema True \
  --log_every 1 \
  --lr 2e-5 \
  --num_worker 1 \
  --expected_num_tokens 4096 \
  --max_num_tokens 5120 \
  --max_num_tokens_per_sample 4096 \
  --cpu_offload True
```
感谢！

**Comment by Koinetivc** [comment]
你好，请问你解决这个问题了吗，我也碰到了这个。我是4张卡，200G的内存会爆掉，如果只用两张卡就可以成功加载进来，后面把模型放到卡上的时候内存占用也降低的非常快，但是我两张卡放不下模型。

**Comment by liuxuexun** [comment]
还没解决，我甚至没试过用两张卡，我回去试试

> 你好，请问你解决这个问题了吗，我也碰到了这个。我是4张卡，200G的内存会爆掉，如果只用两张卡就可以成功加载进来，后面把模型放到卡上的时候内存占用也降低的非常快，但是我两张卡放不下模型。

**Comment by FrankYang-17** [comment]
@liuxuexun Hi, did u solve it?

**Comment by liuxuexun** [comment]
> [@liuxuexun](https://github.com/liuxuexun) Hi, did u solve it?
You can try to use 2 cards, but I haven't try yet.

**Comment by FrankYang-17** [comment]
Thank you for your reply!
I tried 2 cards, but the finetune speed is too slow, so I’m trying other solutions...




---- Replied Message ----
| From | ***@***.***> |
| Date | 07/21/2025 11:26 |
| To | ***@***.***> |
| Cc | Yang ***@***.***> ,
***@***.***> |
| Subject | Re: [ByteDance-Seed/Bagel] OOM (Issue #172) |
liuxuexun left a comment (ByteDance-Seed/Bagel#172)

@liuxuexun Hi, did u solve it?
You can try to use 2 cards, but I haven't try yet.
